### ISSUE TITLE
Remove compatibility Jython code

### ISSUE THREAD

**Issue by sobolevn** [python/cpython#99482]
There are several places where some Jython-specific hacks are used. For example, there are imports from `com.python.core` which is a third-party library.

They are untested and undocumented.

Right now Jython is stuck with 2.7
I think, we don't have to keep it: eventually Jython can patch it on their side, since all parts are pure-Python code.

Original issue on Jython's repo: https://github.com/jython/jython/issues/217

I think we should leave this open for some time (1 month?) to let Jython's team to comment on this.

CC @Yhg1s 

PR to review and comment is incoming :)

### Linked PRs
* gh-99484

**Comment by vstinner** [comment]
PyPy has patches on top of CPython stdlib. They fork the stdlib. I would prefer to avoid "downstream patches", but I don't recall any attempt from PyPy to merge these changes in CPython upstream.

IMO it's ok if Jython also has "downstream patches" on top of CPython stdlib. I expect that they need a few changes to better integrate the stdlib with Jython, optimize the code, give access to Jython features, etc.

Sadly yes, Jython seems inactive and I also would prefer to not continue maintaining code which is specific to Jython if nobody recalls where it comes from, if it's not tested, etc.

**Comment by sobolevn** [comment]
There's also https://docs.python.org/3/library/platform.html#java-platform
But, I don't think we can touch it, since it is a public API.

**Comment by sobolevn** [comment]
Citing Jython team:

> tl;dr: It's fine to remove from 3.x; I assume you have no interest in doing so from 2.7.

Link: https://github.com/jython/jython/issues/217#issuecomment-1314368678
